test/docs/dash-test-framework.md

```diff
@@ -0,0 +1,54 @@
+
+# **DASH Testing Framework**
+
+Here we describe a DASH testing framework that can be utilized for two main purposes:
+1. To run highly scaled testing scenarios for SAI DASH testing
+2. To write API agnostic tests to allow same script to run on multiple supported target APIs
+
+## **Overview**
+
+This testing framework was needed to mainly address the challenge of writing scaled tests to determine the limits of physical and virtual SAI implementations. SAI-challenger was used as the base framework. Below are some of the salient features of this current framework.
+
+1. Support DPU as a target
+2. DUT API abstraction 
+3. Support of Thrift API in SAI Challenger
+4. Multiple test config format
+5. Support of SAI config gen format
+6. Thrift agent refactoring to support all types of SAI object
+7. More efficient way to convert SAI types
+8. Added snappi way for flow definitions (packets formats, incremented fields)
+
+## **High-level Diagram**
+
+![High-level Diagram](../../images/confgen-hld-diag-ref.svg)
+
+The generators produce Python data structures which can be rendered into output text (e.g. JSON) or used to feed custom applications such as a saithrift API driver, to directly configure a device. Likewise a custom API driver can be developed for vendor-specific APIs.
+
+## **Detailed Diagram**
+
+The following diagram reproduces the detailed inner structure of saigen and shows how a testcase can utilize the generator as imported Python modules to turbo-charge test-cases.
+
+![Detailed Diagram](../../images/confgen-saichallenger-keys.svg)
+
+
+## **Test Setup Details**
```

Comment:
I'm not sure providing a bunch of links to another location is useful. We should just refer people to SAI Challenger repo and provide some high-level context into how it is used. Try to avoid use of URLs pointing to dev branches which are going to disappear once a PR is merged.

Reply:
Yes - this warrants a separate document and perhaps bit more context. The idea was to have one placeholder to capture all related information in one place.  i can leave this entire file out of the PR - if it makes sense.